File: 00_variables.f90
program main
    implicit none
    ! Variable declarations first
    integer :: integerVar = 123
    logical :: true = .true., false = .false.
    
    real :: realVar = 1.1234567890123456789
    real(4) :: singlePVar = 1.1234567890123456789
    real(8) :: doublePVar = 1.1234567890123456789D0

    character(len=14) :: str = "Hello, Fortran"
    character(len=:), allocatable :: dynamicStr
    
    ! Integers and bool
    PRINT*, integerVar
    PRINT*, true, false
    WRITE(*, *) integerVar
    WRITE(*, '(I12)') integerVar
    WRITE(*, '(I5)') integerVar
    
    ! Floats
    ! field width of 20 and 16 decimal places
    WRITE(*, '(F20.16)') realVar
    WRITE(*, '(F20.16)') singlePVar
    WRITE(*, '(F20.16)') doublePVar
    
    ! Strings
    WRITE(*, '(A)') str

    ! Assigning a value to the allocatable string
    dynamicStr = "Hello, Fortran"
    WRITE(*, "(A,T22,A)") "Initial string: ", dynamicStr
    ! Modifying the string (slicing and concatenation)
    dynamicStr = dynamicStr(1:7) // "World!!!!!!!!!!"
    WRITE(*, "(A,T22,A)") "After modification: ", dynamicStr

end program main

File: 00_variables.cpp
#include <iostream>
#include <cstdio>
using namespace std;

int main() {
    // integers and bool
    int integerVar = 123;
    bool trueVar = true, falseVar = false;
        // Printing using cout or printf
    cout << integerVar << endl;
    cout << trueVar << " " << falseVar << endl;
    printf("%d\n", integerVar);
    printf("%d %d\n", trueVar, falseVar);
    
    // floats
    float singlePVar = 1.1234567890123456789;
    double doublePVar = 1.1234567890123456789;
    printf("%20.16f\n", singlePVar);
    printf("%20.16f\n", doublePVar);

    // Strings
    char str[11] = "Hello, C++";
    printf("%s\n", str);

    string dynamicStr = "Hello, C++";
    printf("Initial string: \t%s\n", dynamicStr.c_str());
    dynamicStr = "Hello, World!!!!!!!!!!";
    printf("After modification: \t%s\n", dynamicStr.c_str());

    return 0;
}
